core/src/com/unciv/logic/map/TileMap.kt

```diff
@@ -119,6 +122,18 @@ class TileMap {
         return tilesToReturn
     }
 
+    /**
+     * returns the tiles that the parameter is sharing a river with or
+     * an empty list if there is no river next to it.
+     */
+    fun getRiverBuddies(tile: Vector2): List<Vector2> {
+        return rivers
+                .flatMap { river -> river.course }
+                .filter { riverTilePair -> riverTilePair.toList().contains(tile) }
+                .flatMap { riverTilePair -> riverTilePair.toList() }
+                .filter { riverTile -> (riverTile != tile) }
+    }
```

Comment:
looks like a lot of operations and the calls to this method does a search that returns after first hit anyway, so this is the perfect place to use sequences.  Also, maybe I am not following the logic correctly, but this seems super complicated for simply filtering for all the Vector2's that are not equal to the tile, no?  Or is it doing something else?

```kotlin
return rivers.asSequence()
        .flatMap { river -> river.course.asSequence() }
        .filter { riverTilePair -> riverTilePair.first == tile || riverTilePair.second == tile }
        .flatMap { riverTilePair -> sequenceOf(riverTilePair.first, riverTilePair.second) }
        .filterNot { riverTile -> riverTile == tile }.toList()

Reply:
If the goal is what I think, I think it could be simplified to:
```kotlin
return rivers.asSequence().flatMap { 
        river -> river.course.asSequence().map { 
             riverTilePair -> if (riverTilePair.first == tile) riverTilePair.second
                    else if (riverTilePair.second == tile) riverTilePair.first else null
        }.filterNotNull() 
}

Reply:
Also, if each river can have at most 6 or 7 or something river buddies, then you could take on a Take(6) or whatever to the end of the filterNotNull() or even before that, though that's just a very minor optimization since i don't think it's always 6.  If it's less, you still have to iterate through the whole collection with Take(6)